src/datomish/db.cljc

```diff
@@ -130,22 +148,31 @@
         ;; TODO: batch insert, batch delete.
         (doseq [datom datoms]
           (let [[e a v tx added] datom
-                v (ds/->SQLite (.-schema db) a v)] ;; TODO: understand why (schema db) fails.
+                schema (.-schema db) ;; TODO: understand why (schema db) fails.
+                v (ds/->SQLite schema a v)
+                fulltext? (ds/fulltext? schema a)]
             ;; Append to transaction log.
             (<? (exec
                   ["INSERT INTO transactions VALUES (?, ?, ?, ?, ?)" e a v tx (if added 1 0)]))
             ;; Update materialized datom view.
             (if (.-added datom)
-              (<? (exec
-                    ["INSERT INTO datoms VALUES (?, ?, ?, ?, ?, ?, ?, ?)" e a v tx
-                     (ds/indexing? (.-schema db) a) ;; index_avet
-                     (ds/ref? (.-schema db) a) ;; index_vaet
-                     (ds/unique-value? (.-schema db) a) ;; unique_value
-                     (ds/unique-identity? (.-schema db) a) ;; unique_identity
-                     ]))
-              (<? (exec
-                    ;; TODO: verify this is correct.
-                    ["DELETE FROM datoms WHERE (e = ? AND a = ? AND v = ?)" e a v]))))))
+              (let [v (if fulltext?
+                        (<? (<insert-fulltext-value db v))
+                        v)]
+                (<? (exec
+                      ["INSERT INTO datoms VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)" e a v tx
+                       (ds/indexing? schema a) ;; index_avet
+                       (ds/ref? schema a) ;; index_vaet
+                       fulltext? ;; index_fulltext
+                       (ds/unique-value? schema a) ;; unique_value
+                       (ds/unique-identity? schema a) ;; unique_identity
+                       ])))
+              (if fulltext?
+                (<? (exec
+                      ;; TODO: in the future, purge fulltext values from the fulltext_datoms table.
```

Comment:
You did file the issue (https://github.com/mozilla/datomish/issues/19) and I'd prefer to leave it as is for now.  Excision is a future issue: https://github.com/mozilla/datomish/issues/21.